dev-client/src/components/menus/MenuItem.tsx

```diff
@@ -0,0 +1,158 @@
+/*
+ * Copyright © 2024 Technology Matters
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published
+ * by the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program. If not, see https://www.gnu.org/licenses/.
+ */
+
+import {useMemo} from 'react';
+import {PressableProps, StyleSheet, View} from 'react-native';
+import {Text, TouchableRipple} from 'react-native-paper';
+
+import {Icon, IconName} from 'terraso-mobile-client/components/icons/Icon';
+import {convertColorProp} from 'terraso-mobile-client/components/util/nativeBaseAdapters';
+
+export type MenuItemVariant = 'default' | 'destructive';
+
+export type MenuItemProps = {
+  variant?: MenuItemVariant;
+  uppercase?: boolean;
+  label: string;
+  subLabel?: string;
+  icon?: IconName | React.ReactElement;
+  pill?: React.ReactNode;
+  selected?: boolean;
+  disabled?: boolean;
+  onPress?: PressableProps['onPress'];
+};
+
+export function MenuItem({
+  variant = 'default',
+  uppercase = false,
+  icon,
+  label,
+  subLabel,
+  pill,
+  selected,
+  disabled,
+  onPress,
+  children,
+}: React.PropsWithChildren<MenuItemProps>) {
+  const variantStyle = useMemo(() => {
+    if (disabled) {
+      return styles.disabled;
+    } else if (variant === 'destructive') {
+      return styles.destructive;
+    } else {
+      return styles.default;
+    }
+  }, [disabled, variant]);
+
+  return (
+    <TouchableRipple
+      onPress={onPress!}
```

Comment:
Question: What guarantees `onPress` being non-null / non-undefined, since it's marked optional in the props?